Directory.Build.props

```diff
@@ -1,3 +1,7 @@
 <Project>
   <Import Project="Version.props" />
+  <PropertyGroup>
+    <!-- Fail CI warning FS3511: This state machine is not statically compilable.-->
+    <WarningsAsErrors>FS3511</WarningsAsErrors>
```

Comment:
I prefer using `TreatWarningsAsErrors`, as that treats all warnings as errors. I should have had that on long time ago. I'll make the change here.